plugins/chat/lib/chat_message_creator.rb

```diff
@@ -81,6 +90,56 @@ def validate_channel_status!
     end
   end
 
+  def validate_reply_chain!
+    return if @in_reply_to_id.blank?
+
+    @root_message_id = DB.query_single(<<~SQL).last
+      WITH RECURSIVE root_message_finder( id, in_reply_to_id )
+      AS (
+        -- start with the message id we want to find the parents of
+        SELECT id, in_reply_to_id
+        FROM chat_messages
+        WHERE id = #{@in_reply_to_id}
+
+        UNION ALL
+
+        -- get the chain of direct parents of the message
+        -- following in_reply_to_id
+        SELECT cm.id, cm.in_reply_to_id
+        FROM root_message_finder rm
+        JOIN chat_messages cm ON rm.in_reply_to_id = cm.id
+      )
+      SELECT id FROM root_message_finder
+
+      -- this makes it so only the root parent ID is returned, we can
+      -- exclude this to return all parents in the chain
+      WHERE in_reply_to_id IS NULL;
+    SQL
+
+    raise StandardError.new(I18n.t("chat.errors.root_message_not_found")) if @root_message_id.blank?
+
+    @root_message = ChatMessage.with_deleted.find_by(id: @root_message_id)
```

Comment:
sometimes we use `original_message`, sometimes `root_message`, I would settle on `original_message` everywhere

Reply:
Yep good point will fix